addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_animation_sampler_keyframes.py

```diff
@@ -203,6 +203,10 @@ def gather_keyframes(blender_object_if_armature: typing.Optional[bpy.types.Objec
 
         start_frame = min([channel.range()[0] for channel in channels  if channel is not None])
         end_frame = max([channel.range()[1] for channel in channels  if channel is not None])
+
+        if export_settings[gltf2_blender_export_keys.FRAME_RANGE]:
+            start_frame = max(bake_range_start, start_frame)
+            end_frame = min(bake_range_end, end_frame)
```

Comment:
Why using frame range only when fcurve is not sampled?

Reply:
No reason apart from general unfamiliarity with the code, which made me err on the side of minimizing the change.